lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/TestJsonUtils.java

```diff
@@ -70,4 +73,24 @@ public void testTrailingContent()
                 .hasMessage("Could not parse JSON")
                 .hasStackTraceContaining("Unrecognized token 'not': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')");
     }
+
+    @Test
+    public void factoryHasNoReadConstraints()
+    {
+        assertReadConstraints(jsonFactory().streamReadConstraints());
+        assertReadConstraints(JsonUtils.jsonFactoryBuilder().build().streamReadConstraints());
```

Comment:
When this assertion fails, what should one do?
add code comment explaining why it matters / linking to explanation (eg link to issue)